Logger/BaseLoggerMixins.cs

```diff
@@ -1,7 +1,38 @@
-﻿namespace Logger
+﻿using System;
+
+namespace Logger
 {
     public static class BaseLoggerMixins
     {
-        
+        public static void Error(this BaseLogger logger, string message, params Object[] arguments)
+        {
+            if (logger is null) throw new ArgumentNullException(nameof(logger));
+
+            logger.Log(LogLevel.Error, string.Format(message, arguments));
+        }
+
+        public static void Warning(this BaseLogger logger, string message, params Object[] arguments)

```

Comment:
```suggestion
        public static void Warning(this BaseLogger logger, string message, params object[] arguments)

```
Again, you can simplify this to just use the C# keyword rather than the BCL name. You can refer to the guidelines on page 47 in chapter 2 of the book for more information about this. This goes for all of the other methods in this file as well.